### PR TITLE
Changed importlib_metadata to have requirement for < python 3.10

### LINKED ISSUE #575: Make importlib_metadata optional on Python 3.10.
> That extra dependency will become optional again in Python 3.10 alpha 7.

Could it be marked as required only for Python < 3.10 in `setup.cfg`?

_Originally posted by @dotlambda in https://github.com/jaraco/keyring/issues/503#issuecomment-1134170726_

### PR DESCRIPTION
**Pull request by probro27**
Trying to fix the issue #575.
Hope it works!!

### CODE REVIEW

File: setup.cfg
Comment (mitya57) [inline setup.cfg]: We should still specify the minimum version of `importlib_metadata`, and I don't see the point of specifying the maximum version (the latest version of backport is 4.x and it should work fine).

**Review by mitya57 — CHANGES_REQUESTED** [review]
Also please see what I wrote in this comment:
https://github.com/jaraco/keyring/pull/576#pullrequestreview-987281982

File: setup.cfg
Comment (probro27) [reply setup.cfg]: I was just trying to solve the open issue. I'm not super familiar with this. What I got was that importlib_metadata was kind of increasing the dependencies that may not be required for python versions over 3.10. 
If you want I can put importlib_metadata to be compolusory for python >=3.6 and <= 3.10. 
Hope this works out! Just starting out with open source so hoping it's fine.

Comment (mitya57) [reply setup.cfg]: The correct line would be:
```suggestion
	importlib_metadata >= 3.6; python_version < "3.10"
```

This means: require `importlib_metadata` at least 3.6 for Python older than 3.10.

Comment (probro27) [reply setup.cfg]: Oh I understand now! Thank you! I'll commit this suggestion so that it's added to the pull request!
thanks for the tip. First time in open source so was really lost!
Appreciate it!

Comment (mitya57) [reply setup.cfg]: Now please also see this: https://github.com/jaraco/keyring/pull/576#pullrequestreview-987281982

It needs to be replaced in three files: `hook-keyring.backend.py`, `keyring/backend.py`, `tests/test_packaging.py`.

**Comment by probro27** [comment]
I have updated the imports as per [#576 (review)](https://github.com/jaraco/keyring/pull/576#pullrequestreview-987281982)
Hope this works!

File: tests/test_packaging.py
Comment (mitya57) [inline tests/test_packaging.py]:
```suggestion
    import importlib_metadata as metadata
```

**Review by mitya57 — CHANGES_REQUESTED** [review]

**Review by mitya57 — APPROVED** [review]

**Comment by probro27** [comment]
Hmm, any idea what might cause the tests to fail? I  didn't change anything that could cause errors. I'll try to investigate. If you find anything, let me know! 
Thanks!

**Comment by mitya57** [comment]
So mypy complains about this:
```
______________________________ keyring/backend.py ______________________________
15: error: Name "metadata" already defined (by an import)
___________________________ tests/test_packaging.py ____________________________
4: error: Name "metadata" already defined (by an import)
```

I would simply silence that. Try adding `# type: ignore` comments at the end of line 15 of `keyring/backend.py` and line 4 of `tests/test_packaging.py`.

Like this:
```python
try:
    from importlib import metadata
except ImportError:
    import importlib_metadata as metadata  # type: ignore
```

**Comment by probro27** [comment]
Alright I have added that. Thanks for the help! 
Let's hope it works now!

**Comment by mitya57** [comment]
I am sorry. It still doesn't work because Python 3.9 and earlier does have `importlib.metadata` available, although it does not support `group` argument, so we can't use it.

Let's try this:

```python
import sys

if sys.version_info >= (3, 10):
    from importlib import metadata
else:
    import importlib_metadata as metadata
```

Add `import sys`

I hope `# type: ignore` won't be needed with this, but let's see.

**Comment by probro27** [comment]
I have added the `# type:ignore ` on the imports for now, coz I was getting a linting error when I was not doing that. I hope this works!

**Comment by probro27** [comment]
`_________________________________ FLAKE8-check _________________________________
/home/runner/work/keyring/keyring/tests/test_packaging.py:6:42: E261 at least two spaces before inline comment
/home/runner/work/keyring/keyring/tests/test_packaging.py:6:43: E262 inline comment should start with '# '`
Does this seem to be the error?

**Comment by mitya57** [comment]
Yes, you need two spaces before `#` and one space after `#`. And please remove commented out code.

**Comment by probro27** [comment]
Okay, I have changed the formatting and removed the comments.

**Review by mitya57 — APPROVED** [review]
Yes, now CI finally passes. Thanks!

You have my approval, but let's wait for @jaraco who is the chief maintainer.

**Review by jaraco — APPROVED** [review]
I added a couple of nitpicky changes and a changelog, but this looks good. Thanks for pulling it together!

**Comment by probro27** [comment]
Thanks for merging it! 
Glad to be of help :))

**Pull request merged**
